sdk/servicebus/azure-servicebus/azure/servicebus/management/_models.py

```diff
@@ -193,3 +236,498 @@ def _from_internal_entity(cls, internal_qr):
+        self._internal_td.requires_duplicate_detection = self.requires_duplicate_detection
+        self._internal_td.duplicate_detection_history_time_window = self.duplicate_detection_history_time_window
+        self._internal_td.enable_batched_operations = self.enable_batched_operations
+        self._internal_td.size_in_bytes = self.size_in_bytes
+        self._internal_td.is_anonymous_accessible = self.is_anonymous_accessible
+        self._internal_td.authorization_rules = self.authorization_rules
+        self._internal_td.status = self.status
+        self._internal_td.support_ordering = self.support_ordering
+        self._internal_td.auto_delete_on_idle = self.auto_delete_on_idle
+        self._internal_td.enable_partitioning = self.enable_partitioning
+        self._internal_td.entity_availability_status = self.entity_availability_status
+        self._internal_td.enable_subscription_partitioning = self.enable_subscription_partitioning
+        self._internal_td.enable_express = self.enable_express
+        self._internal_td.user_metadata = self.user_metadata
+
+        return self._internal_td
+
+
+class TopicRuntimeInfo(object):
+    """Description of a Service Bus topic resource.
+
+    """
+    def __init__(
+        self,
+        **kwargs
+    ):
+        """
+        :keyword str name:
+        :keyword created_at: The exact time the queue was created.
+        :type created_at: ~datetime.datetime
+        :keyword updated_at: The exact time a message was updated in the queue.
+        :type updated_at: ~datetime.datetime
+        :keyword accessed_at: Last time a message was sent, or the last time there was a receive request
+         to this queue.
+        :type accessed_at: ~datetime.datetime
+        :keyword message_count_details: Details about the message counts in queue.
+        :type message_count_details: ~azure.servicebus.management._generated.models.MessageCountDetails
+        :keyword subscription_count: The number of subscriptions in the topic.
+        :type subscription_count: int
+
+        :rtype: None
+        """
+        self.name = kwargs.get('name', None)
+        self._internal_td = None
+        self.created_at = kwargs.get('created_at', None)
+        self.updated_at = kwargs.get('updated_at', None)
+        self.accessed_at = kwargs.get('accessed_at', None)
+        self.message_count_details = kwargs.get('message_count_details', None)
+        self.subscription_count = kwargs.get('subscription_count', None)
+
+    @classmethod
+    def _from_internal_entity(cls, internal_td):
+        # type: (InternalTopicDescription) -> TopicRuntimeInfo
+        qd = cls()
+        qd._internal_td = internal_td
+
+        qd.created_at = internal_td.created_at
+        qd.updated_at = internal_td.updated_at
+        qd.accessed_at = internal_td.accessed_at
+        qd.message_count_details = internal_td.message_count_details
+        qd.subscription_count = internal_td.subscription_count
```

Comment:
This pattern makes me feel a little uncomfortable.  I won't have us rethink this this late in the game, but ESPECIALLY given that we have those "workaround" enumerations to give arg order, a lot of me wonders if we'd be better suited having all this property-connectivity being automated.

Take this question as: "if we eventually need to reapproach this or add more types, is there a reason that we did this all manually out as opposed to metaprogramming?"  (For my own education in the future; again, I don't want to have us rewrite this now)